### PR TITLE
Implement MTA-STS and DANE validation

### LINKED ISSUE #707: Verify DANE when sending emails and use STARTTLS when possible.
According to https://ssl-tools.net/mailservers/jcg.re, mailu seems to not use StartTLS when sending emails. Additionally, https://havedane.net/ indicates that when a DANE record is available but an invalid certificate is used, mailu will still deliver the mail. It makes sense, since it doesn't use StartTLS there is no certificate to validate.

### LINKED ISSUE #1798: [enhancement-requests] mta-sts support?
Thank you for opening an issue with Mailu. Please understand that issues are meant for bugs and enhancement-requests.
For **user-support questions**, reach out to us  on [matrix](https://matrix.to/#/#mailu:tedomum.net).

image
my mail server gets blocked by Outlook. so it would be great to have an mta-sts option to prove the e-mail server.
I'm not a professional on the subject. With Mailu I managed to install an email server for the first time.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

Feature

## What does this PR do?

Implement MTA-STS: the tls_policy_map will now be auto-configured based on the policies published by the various domains. A FAQ entry has been added to document how to publish a policy using Mailu.

As configured by default there is no persistence. If we want persistence we can have either sqlite3 (with a db in the mailqueue) or redis...

This also introduces a DEFER_ON_TLS_ERROR (default: True) setting that will harden policy enforcement and defer emails that shouldn't be delivered. Turn it off if you never want to set an override.

### Related issue(s)
- closes #1798
- closes #707 

## Prerequistes
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
@Diman0 @ofthesun9 For context on what this tries to achieve:
https://github.com/internetstandards/toolbox-wiki/blob/master/DANE-for-SMTP-how-to.md
https://github.com/Snawoot/postfix-mta-sts-resolver/issues/67

File: core/admin/mailu/utils.py [hunk 1]
```diff
@@ -37,6 +39,28 @@ def handle_needs_login():
...
 
+# DNS stub configured to do DNSSEC enabled queries
+resolver = dns.resolver.Resolver()
+resolver.use_edns(0, 0, 1232)
+resolver.flags = dns.flags.AD | dns.flags.RD
+
+def has_dane_record(domain, timeout=10):
+    try:
+        result = resolver.query(f'_25._tcp.{domain}', dns.rdatatype.TLSA,dns.rdataclass.IN, lifetime=timeout)
+        if (result.response.flags & dns.flags.AD) == dns.flags.AD:
```
Comment (ghostwheel42) [inline core/admin/mailu/utils.py:50]: what about skipping the == part:
`if result.response.flags & dns.flags.AD:`

**Comment by nextgens** [comment]
bors try

**Review by ghostwheel42 — APPROVED** [review]
lgtm

**Pull request merged**